src/core/server/saved_objects/service/lib/filter_utils.ts

```diff
@@ -128,7 +128,7 @@ export const validateFilterKueryNode = (
   }, []);
 };
 
-const getType = (key: string) => (key.includes('.') ? key.split('.')[0] : null);
+const getType = (key: string) => (key != null && key.includes('.') ? key.split('.')[0] : null);
```

Comment:
NIT: I would think that it's more of the caller's responsibility to ensure the passed string is not null (same goes for `isSavedObjectAttr`)?

`type: getType(ast.value),` => `type:typeof ast.value === 'string' ? getType(ast.value) : null`

Or at least change signature to `string | undefined/null` as in `isSavedObjectAttr `

Reply:
I agree. Another way could be to use `hasFilterKeyError()` like validation, if that fails we really don't care about running `getType` or `isSavedObjectAttr`:
```
const filterKeyError = hasFilterKeyError();
return [
  ...kueryNode,
  {
    astPath: splitPath.slice(0, splitPath.length - 1).join('.'),
    error: filterKeyError,
    isSavedObjectAttr: filterKeyError != null ? isSavedObjectAttr(ast.value, indexMapping) : null,
    key: ast.value,
    type: filterKeyError != null ? getType(ast.value) : null,
  },
];
```

Reply:
You just have to be careful as the `ast.value` also would be null/undefined for isSavedObjectAttr and fail there as well. I will fix the typings suggested above and check-in.

However feel free to change the logic around if you think its better to pull the logic up into a filter like mechanism. I am not as familiar with the `ast.value` and what it should allow or not allow as you point out below it is any `any` right now.